util/src/main/java/org/datacommons/util/ReconClient.java

```diff
@@ -49,6 +56,42 @@ public ReconClient(HttpClient httpClient, LogWrapper logWrapper, int chunkSize)
     this.chunkSize = chunkSize;
   }
 
+  public ResolveResponse resolve(ResolveRequest request) {
+    try {
+      return resolveAsync(request).get();
+    } catch (Exception e) {
+      throw new RuntimeException("Error resolving nodes.", e);
+    }
+  }
+
+  public CompletableFuture<ResolveResponse> resolveAsync(ResolveRequest request) {
+    ResolveResponse defaultResponse = ResolveResponse.getDefaultInstance();
+    if (request.getNodesCount() < 1) {
+      return CompletableFuture.completedFuture(defaultResponse);
+    }
+
+    return toFutureOfList(
+            // Partition request into chunkSize batches.
```

Comment:
Curious, is this a parallel call for different chunks?

Reply:
Yes it is. It roughly works like this:

```
Request -> partition -> 
List<Request> -> call DC in parallel for each (chunked) Request -> List<Future<Response>> -> 
Future<List<Response>> -> join -> Future<Response>
```

I'll perhaps do a PR to capture these states into named variables so it's easier to follow.